Authentication events policy docs for user flow invocation (#10224)
microsoftgraph/microsoft-graph-docs: api-reference/beta/resources/invokeuserflowaction.md

|priority|Int32|The priority of the action which is used for determining one out of multiple applicable actions. Inherited from [authenticationAction](../resources/authenticationaction.md)|
|sourceFilter|[authenticationSourceFilter](../resources/authenticationsourcefilter.md)|Filter based on the source of the authentication which is used to determine whether the action is executed or not. Inherited from [authenticationAction](../resources/authenticationaction.md)|

## Relationships

|Relationship|Type|Description|
|:---|:---|:---|
|userFlow|[b2xIdentityUserFlow](../resources/b2xidentityuserflow.md)|The user flow that will be invoked if this action is executed.|

## JSON representation

The following is a JSON representation of the resource.
<!-- {
  "blockType": "resource",
  "keyProperty": "id",
  "@odata.type": "microsoft.graph.invokeUserFlowAction",
  "baseType": "microsoft.graph.authenticationAction",
  "openType": false
}
-->

``` json
{
  "@odata.type": "#microsoft.graph.invokeUserFlowAction",
  "id": "String (identifier)",
  "priority": "Integer",
  "sourceFilter": {
    "@odata.type": "microsoft.graph.authenticationSourceFilter"
  },

>   "userFlow": "String"

Reviewer: I just noticed here that the userFlow should be an object and not a string. It should be represented like your other examples:
    "userFlow": {
        "id": "B2X_1_Partner"
    }

Author: Good catch. Fixed

>   "userFlow": "String"

Author: For the resource files, yes.

---
title: "invokeUserFlowAction resource type"
description: "An action that is used to invoke a user flow during an authentication event."
author: "jkdouglas"
localization_priority: Normal
ms.prod: "microsoft-identity-platform"
doc_type: resourcePageType
---

# invokeUserFlowAction resource type

Namespace: microsoft.graph

An action that is used to invoke a user flow during an authentication event. Authentication events are controlled by the [authenticationEventsPolicy](../resources/authenticationeventspolicy.md).

> Inherits from [authenticationAction](../resources/authenticationaction.md).

Reviewer 2: Line 16:
Perhaps you can further clarify that authenticationAction is an abstract resource? Hopefully that can help clarify for customers, re David's comment how customers could possibly figure out why the "create", "update", "put" operations of authenticationAction would include the userFlow property in the request body section.

Something like:
"Inherits from the abstract base type authenticationAction."

## Properties

|Property|Type|Description|
|:---|:---|:---|
|id|String|The identifier of the action. Inherited from [authenticationAction](../resources/authenticationaction.md)|

> |priority|Int32|The priority of the action which is used for determining one out of multiple applicable actions. Inherited from [authenticationAction](../resources/authenticationaction.md)|

Reviewer 3: ```suggestion
|priority|Int32|The priority of the action that is used to determine one out of multiple applicable actions. Inherited from [authenticationAction](../resources/authenticationaction.md)|
```

> |sourceFilter|[authenticationSourceFilter](../resources/authenticationsourcefilter.md)|Filter based on the source of the authentication which is used to determine whether the action is executed or not. Inherited from [authenticationAction](../resources/authenticationaction.md)|

Reviewer 3: ```suggestion
|sourceFilter|[authenticationSourceFilter](../resources/authenticationsourcefilter.md)|Filter based on the source of the authentication that is used to determine whether the action is executed. Inherited from [authenticationAction](../resources/authenticationaction.md)|
```

## Relationships

|Relationship|Type|Description|
|:---|:---|:---|

> |userFlow|[b2xIdentityUserFlow](../resources/b2xidentityuserflow.md)|The user flow that will be invoked if this action is executed.|

Reviewer 3: ```suggestion
|userFlow|[b2xIdentityUserFlow](../resources/b2xidentityuserflow.md)|The user flow that is invoked if this action is executed.|
```
